python/graphframes/tests.py

```diff
@@ -204,74 +224,93 @@ def test_dropIsolatedVertices(self):
 
     def test_bfs(self):
         g = self.g
         paths = g.bfs("name='A'", "name='C'")
-        self.assertEqual(paths.count(), 1)
-        self.assertEqual(paths.select("v1.name").head()[0], "B")
+        assert paths.count() == 1
+        # Expecting that the first intermediary vertex in the BFS is "B"
+        assert paths.select("v1.name").head()[0] == "B"
+
         paths2 = g.bfs("name='A'", "name='C'", edgeFilter="action!='follow'")
-        self.assertEqual(paths2.count(), 0)
+        assert paths2.count() == 0
+
         paths3 = g.bfs("name='A'", "name='C'", maxPathLength=1)
-        self.assertEqual(paths3.count(), 0)
+        assert paths3.count() == 0
 
 
-class PregelTest(GraphFrameTestCase):
-    def setUp(self):
-        super(PregelTest, self).setUp()
+@pytest.mark.usefixtures("set_spark")
+class TestPregel:
 
     def test_page_rank(self):
-        from pyspark.sql.functions import coalesce, col, lit, sum, when
-        edges = self.spark.createDataFrame([[0, 1],
-                                          [1, 2],
-                                          [2, 4],
-                                          [2, 0],
-                                          [3, 4], # 3 has no in-links
-                                          [4, 0],
-                                          [4, 2]], ["src", "dst"])
+        # Create an edge DataFrame; note that vertex 3 has no in-links.
+        edges = self.spark.createDataFrame(
+            [[0, 1], [1, 2], [2, 4], [2, 0], [3, 4], [4, 0], [4, 2]],
+            ["src", "dst"],
+        )
         edges.cache()
+
+        # Create a vertex DataFrame and count vertices.
         vertices = self.spark.createDataFrame([[0], [1], [2], [3], [4]], ["id"])
         numVertices = vertices.count()
+
+        # Get the outDegrees DataFrame from a GraphFrame built on the original vertices and edges.
         vertices = GraphFrame(vertices, edges).outDegrees
         vertices.cache()
+
+        # Construct a new GraphFrame with the updated vertices DataFrame.
         graph = GraphFrame(vertices, edges)
         alpha = 0.15
-        ranks = graph.pregel \
-            .setMaxIter(5) \
-            .withVertexColumn("rank", lit(1.0 / numVertices),
-                              coalesce(Pregel.msg(),
-                                       lit(0.0)) * lit(1.0 - alpha) + lit(alpha / numVertices)) \
-            .sendMsgToDst(Pregel.src("rank") / Pregel.src("outDegree")) \
-            .aggMsgs(sum(Pregel.msg())) \
+
+        # Run PageRank via Pregel.
+        ranks = (
+            graph.pregel.setMaxIter(5)
+            .withVertexColumn(
+                "rank",
+                F.lit(1.0 / numVertices),
+                F.coalesce(Pregel.msg(), F.lit(0.0)) * F.lit(1.0 - alpha)
+                + F.lit(alpha / numVertices),
+            )
+            .sendMsgToDst(Pregel.src("rank") / Pregel.src("outDegree"))
+            .aggMsgs(F.sum(Pregel.msg()))
             .run()
+        )
```

Comment:
```suggestion
        pregel = graph.pregel
        ranks = (
            pregel.setMaxIter(5)
            .withVertexColumn(
                "rank",
                F.lit(1.0 / numVertices),
                F.coalesce(pregel.msg(), F.lit(0.0)) * F.lit(1.0 - alpha)
                + F.lit(alpha / numVertices),
            )
            .sendMsgToDst(pregel.src("rank") / pregel.src("outDegree"))
            .aggMsgs(F.sum(pregel.msg()))
            .run()
        )
```

Because with support of SparkConnect there will be two possible types (Pregel classic and Pregel connect)

Reply:
Done.